### DISCUSSION TITLE
actionban, how to add a comment?

### DISCUSSION THREAD

**Discussion by startup-001-steve** [fail2ban/fail2ban#3481] [Q&A] [answered]
hello,

(I originally posted this to the listserve, then I discovered this is a better place to post it.)

running fail2ban version 0.9.3 on ubuntu 

I want to learn how to add a comment when banning an ip, and have that comment include data / information from the log file f2b is monitoring, using variables that f2b captures in the regex. I tried to read various faqs, webpages, and github, and even searched thru source code, but I only found 1 instance of a comment in ufw.conf

for example, in a log file made by **mail/courier/imap**, it contains this line:
 `  imapd: LOGIN FAILED, user=[cowseight8@domain.com](mailto:cowseight8@domain.com), ip=[::ffff:183.157.169.196]`

in jail.conf,  it appears the default banaction script is "iptables-multiport"
so I created a file in/etc/fail2ban/action.d
named    **iptables-multiport.local**

and pasted this information (found in the FAQ section of f2b user guide website).

`actionban = lgm=$(printf '%%.1000s\n...' "<matches>"); <iptables> -I f2b-<name> 1 -s <ip> -j <blocktype> "comment=$lgm" --comment "$lgm"
`

then I restarted f2b      **systemctl restart fail2ban**

journalctl -ru fail2ban    showed it restarted without errors

how can I check if this is working? will it be written in the iptables, or will it be written in the fail2ban log, or will it be written in /var/log/ufw.log  ?

after I find out where I can see the results, I will probably need to come back and learn how to capture information from the source log, trying to understand which variable names are created by f2b, and how to access them, so my comments inserted in the log help me evaluate security.

thank you

**Comment by sebres** [comment, marked as answer]
> how to add a comment when banning an ip

Add the comment to what exactly?
If you want to add it to some file or database, you can add a second action (and use tag `<matches>` inside) to write them to some file (e. g. [action.d/dummy.conf](https://github.com/fail2ban/fail2ban/blob/master/config/action.d/dummy.conf)) or to database.
I don't think a comment on a netfilter subsystem is a good idea at all.

Why do you need that at all?
One can obtain the matched lines from fail2ban database with `sqlite3` or from monitoring log/journal using `fail2ban-regex` (especially with newer version supporting `-o` parameter).

> lgm=$(printf '%%.1000s\n...' "<matches>"); 

Yes, tag `<matches>` indeed contains the matched log-lines captured by filter (up to `maxmatches`).
For more info see [man jail.conf.5](https://manpages.debian.org/unstable/fail2ban/jail.conf.5.en.html#Action_Tags)

> how can I check if this is working? 

Normally if the command fails, you'd see an error in fail2ban.log, otherwise it does what you specified in the command `actionban`.

> will it be written in the iptables

Conditionally true, so you have to check it with `iptables -nL ...`
However the syntax `comment=$lgm` is unknown to me and I thought `-comment` additionally expects `-m comment`, so I know only format like this:
```
iptables ... -m comment --comment "some comment here" -j ...
```
Also I'm unsure it would correct work with new-lines (by several matched lines), etc.
As well as one'd need to adjust `actionunban` (to remove the rule with a comment, since it'd expect the same arguments as `actionban`).
Anyway your approach is too error-prone for my taste.

> or will it be written in the fail2ban log

No, unless it fails or you'd not increase the loglevel (to DEBUG).
But then one'd write it with `; printf "%s" "<matches>"` or similar.

> or will it be written in /var/log/ufw.log ?

iptables actions has nothing with ufw, they works with iptables only, so the answer is - no.
